invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/customFields/CustomField.js

```diff
@@ -56,7 +56,7 @@ export class CustomField extends Field {
       // internal implementation
       // Note: if i is an array of strings, then we exclude the above as you cannot set
       // a property on a string
-      if (i && typeof value !== "string") value.__key = i;
+      if (i && typeof value !== "string" && typeof value !== "number") value.__key = i;
```

Comment:
can we maybe list here all the primitive types that are not accepting attribute assignment so we don't go over again and again?